core/src/main/java/org/opensearch/sql/calcite/udf/mathUDF/ModFunction.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.calcite.udf.mathUDF;
+
+import org.opensearch.sql.calcite.udf.UserDefinedFunction;
+
+public class ModFunction implements UserDefinedFunction {
+  @Override
+  public Object eval(Object... args) {
+    if (args.length < 2) {
+      throw new IllegalArgumentException("At least two arguments are required");
```

Comment:
make the error message more precise. "Mod function required two arguments, but it has args.length"